any_run/icon_any_run/connection/connection.py

```diff
@@ -0,0 +1,52 @@
+import insightconnect_plugin_runtime
+from .schema import ConnectionSchema, Input
+# Custom imports below
+from insightconnect_plugin_runtime.exceptions import ConnectionTestException, PluginException
+import base64
+from icon_any_run.util.api import AnyRunAPI
+
+
+class Connection(insightconnect_plugin_runtime.Connection):
+
+    def __init__(self):
+        super(self.__class__, self).__init__(input=ConnectionSchema())
+        self.authentication_header = None
+        self.any_run_api = None
+
+    def connect(self, params):
+        self.logger.info("Connect: Connecting...")
+        api_key = params.get(Input.API_KEY)
+        credentials = params.get(Input.CREDENTIALS)
+
+        if not api_key and not credentials:
+            raise ConnectionTestException(
+                cause='No authentication credentials provided in the connection',
+                assistance='Configure the connection with either an API key or username and password and try again.'
+            )
+
+        if api_key and api_key.get("secretKey") and credentials and credentials.get('username'):
+            raise ConnectionTestException(
+                cause='Multiple authentication methods provided.',
+                assistance='Use a single credential method in the connnection, set either API key or username and password and try again.'
+            )
+
+        if api_key and api_key.get("secretKey"):
+            authorization = f'API-Key {api_key.get("secretKey")}'
+        else:
+            authorization = 'Basic ' + base64.encodebytes(
+                (credentials.get('username') +
+                 ":" +
+                 credentials.get('password')).encode()
+            ).decode("utf-8")
+
+        self.any_run_api = AnyRunAPI({
+            'Authorization': authorization.rstrip()
+        }, self.logger)
+
+    def test(self):
+        try:
```

Comment:
It looks like the API returns when the credentials are invalid. 

```
HTTP/1.1 403 Forbidden
{
  "error": true,
  "message": "Wrong authorization data"
}
```

Can we surface an API or user/password error to the user to help indicate the problem using the following based on auth type users provided.

```
raise ConnectionTestException(preset=ConnectionTestException.Preset.API_KEY)
raise ConnectionTestException(preset=ConnectionTestException.Preset.USERNAME_PASSWORD)
```

More details at 

* https://komand.github.io/python/error_handling.html#connection-exceptions
* https://github.com/rapid7/komand-plugin-sdk-python/blob/master/insightconnect_plugin_runtime/exceptions.py#L45

Reply:
E.g. When the trial expires on the Get History action we get this. We should make this easier to read for the customer by displaying more details found from the API

```
Call to Any Run failed {"error":true,"message":"Requires an active subscription to a Hunter plan"} Response was: 403 Client Error: Forbidden for url: https://api.any.run/v1/analysis/?team=False&skip=0&limit=25
```

Reply:
We add this api as template so for other PRs there should be no this kind of errors.